src/lib.rs

```diff
@@ -65,215 +98,2254 @@ pub struct Leser {
+    #[xml(name = b"akkreditierungsnummer", ty = "child")]
+    pub akkreditierungsnummer: Option<String>,
+    #[xml(name = b"kommentarBeauftragteUntersuchungsstelle", ty = "child")]
+    pub kommentar_beauftragte_untersuchungsstelle: Option<String>,
+    #[xml(name = b"kommentarZugelasseneUntersuchungsstelle", ty = "child")]
+    pub kommentar_zugelassene_untersuchungsstelle: Option<String>,
 }
 
-#[derive(Debug, Default, XmlSerialize, XmlDeserialize, Serialize, Deserialize, Tsify)]
-#[tsify(into_wasm_abi, from_wasm_abi)]
-#[xml(root = b"vorgang.transportieren.2010")]
-#[xml(tns(b"xwas", b"xwasser"))]
-pub struct QualityReport {
+/// Klasse für den Transport von ergänzenden Informationen zu den Daten aus dem Register
+/// für Zugelassene Untersuchungsstellen im Falle der Beauftragung einer Untersuchung.
+#[derive(Default, Debug, XmlSerialize, XmlDeserialize, Serialize, Deserialize, Tsify)]
+#[xml(root = b"BeauftragteUntersuchungsstelleType")]
+pub struct BeauftragteUntersuchungsstelleType {
+    #[xml(ns = b"xwas", name = b"zugelasseneUntersuchungsstelle", ty = "child")]
+    pub zugelassene_untersuchungsstelle: ZugelasseneUntersuchungsstelleType,
     #[xml(
-        ns = b"xmlns",
-        name = b"xsi",
-        ty = "attr",
-        value = "http://www.w3.org/2001/XMLSchema-instance"
+        ns = b"xwas",
+        name = b"kommentarBeauftragteUntersuchungsstelle",
+        ty = "child"
+    )]
+    pub kommentar_beauftragte_untersuchungsstelle: Option<String>,
+}
+
+#[derive(Default, Debug, XmlSerialize, XmlDeserialize, Serialize, Deserialize, Tsify)]
+#[xml(root = b"codeGesamtbewertungType")]
+pub struct CodeGesamtbewertungType {
+    #[xml(ns = b"xwas", name = b"code", ty = "child")]
+    pub code: String,
+    #[xml(ns = b"xwas", name = b"name", ty = "child")]
+    pub name: Option<String>,
+}
+
+#[derive(Default, Debug, XmlSerialize, XmlDeserialize, Serialize, Deserialize, Tsify)]
+#[xml(root = b"codeAmtsspracheEUType")]
+pub struct CodeAmtsspracheEUType {
+    #[xml(ns = b"xwas", name = b"code", ty = "child")]
+    pub code: String,
+    #[xml(ns = b"xwas", name = b"name", ty = "child")]
+    pub name: Option<String>,
+}
+
+/// Erweiterung.XML darf nur dazu genutzt werden, weitere (z. B. fachspezifische)
+/// Metadaten zu spezifizieren, deren Übermittlung mit den bereits in XWasser
+/// spezifizierten Metadaten nicht möglich ist. Erweiterung.XML bietet über ein
+/// xs:any-Element die Möglichkeit, mittels Einbindung externer XML-Schemata diese
+/// Metadaten zu spezifizieren. Es können beliebige XML-Schemata mit unterschiedlichen
+/// Namensräumen angegeben werden. Die XML-Schema-Validierung der weiterführenden
+/// Metadaten erfolgt innerhalb der XWasser-Nachricht selbst (prozessContents = "lax").
+#[derive(Default, Debug, XmlSerialize, XmlDeserialize, Serialize, Deserialize, Tsify)]
+#[xml(root = b"erweiterungXMLType")]
+pub struct ErweiterungXmlType {
+    // weiteres xml schema
```

Comment:
Comments (also in German). Stick to English. The whole comments in German - we need to think about it. If we decide we have German-only library - all documentation must be in German. If we decide opposite, we need to provide documentation for both variants. We can use AI to translate German to English.